spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/util/DeploymentPropertiesUtils.java

```diff
@@ -61,40 +56,40 @@ private DeploymentPropertiesUtils() {
 	 * {@code app.[appname].[key]} or {@code deployer.[appname].[key]}.
 	 * Values may themselves contain commas, since the split points will be based upon the key pattern.
 	 *
-	 *
-	 * Logic of this pattern to find boundaries of valid key=value pairs is:
-	 * 1. (?:,{0,1}\s*) is a non-capturing group which takes characters before a key.
-	 *    Needed so that capturing groups later will capture key boundaries correctly.
-	 * 2. ((\.*(\w|\*|-)+)+) is a capturing group finding an actual key. There is a recursive
-	 *    capturing groups inside of it as we assume key is formed from dots and words and
-	 *    we capture those as many time as needed. We get groups 1, 2 and 3 but we only use 1.
-	 * 3. (\s*=\s*) here we find boundary of '=' which may have empty characters around it.
-	 *    This becomes capturing group 4.
-	 * 4. Now we know where is key with group 1 and start of value with group 4.
-	 * 5. Now we can start to do matching. Need to match one ahead so that we
-	 *    know where next key possible is as that is then end of previous value.
-	 * 6. Lastly we have one pair left which is a same case if we only have
-	 *    one pair to parse.
+	 * Logic of parsing key/value pairs from a string is based on few rules and assumptions
+	 * 1. keys will not have commas or equals.
```

Comment:
given the 1st assumption, would it be simpler to split on `=` and then split between value and the next key by splitting on the last comma (since no comma can exist in a key)?